src/sentry/static/sentry/app/components/featureBadge.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import Tag from 'app/views/settings/components/tag';
+import Tooltip from 'app/components/tooltip';
+import space from 'app/styles/space';
+import {t} from 'app/locale';
+import theme from 'app/utils/theme';
+import CircleIndicator from 'app/components/circleIndicator';
+
+type BadgeProps = {
+  type: 'alpha' | 'beta' | 'new';
+  variant?: 'indicator' | 'badge';
+  title?: string | null;
+};
+
+type Props = Omit<React.HTMLAttributes<HTMLDivElement>, keyof BadgeProps> & BadgeProps;
+
+const defaultTitles = {
+  alpha: t('This feature is in alpha and may be unstable'),
+  beta: t('This feature is in beta and may change in the future'),
+  new: t('This is a new feature'),
+};
+
+const labels = {
+  alpha: t('alpha'),
+  beta: t('beta'),
+  new: t('new'),
+};
+
+const FeaturedBadge = ({type, variant = 'badge', title, ...p}: Props) => {
+  const tooltipTitle = title === undefined ? defaultTitles[type] : title;
+
+  return (
+    <div {...p}>
+      <Tooltip title={tooltipTitle} disabled={!tooltipTitle} position="right">
+        <React.Fragment>
+          {variant === 'badge' && <StyledTag priority={type}>{labels[type]}</StyledTag>}
+          {variant === 'indicator' && (
+            <CircleIndicator color={theme.badge[type].indicatorColor} size={8} />
+          )}
+        </React.Fragment>
+      </Tooltip>
+    </div>
+  );
+};
+
+const StyledTag = styled(Tag)`
+  padding: 3px ${space(0.75)};
```

Comment:
has it to be really 3px? maybe you could use space(0.25) or space(0.5)?

Reply:
haha yeah I know. It really just looks best at 3px and IMO at this small of values we have to prioritize visuals over consistent spacing units.